raiden-cli/src/routes/pendingTransfers.ts

```diff
@@ -1,20 +1,61 @@
 import { Router, Request, Response } from 'express';
+import { timer } from 'rxjs';
+import { filter, takeUntil, map, toArray } from 'rxjs/operators';
+import { RaidenTransfer, isntNil } from 'raiden-ts';
+
 import { Cli } from '../types';
+import { validateOptionalAddressParameter } from '../utils/validation';
 
-export function makePendingTransfersRouter(this: Cli): Router {
-  const router = Router();
+enum ApiRole {
+  initiator = 'initiator',
+  mediator = 'mediator',
+  target = 'target,',
+}
 
-  router.get('/', function (_request: Request, response: Response): void {
-    response.status(404).send('Not implemented yet');
-  });
+function transformSdkToApiPendingTransfer(this: Cli, transfer: RaidenTransfer) {
+  let role: ApiRole;
+  if (transfer.direction === 'sent' && transfer.initiator === this.raiden.address)
+    role = ApiRole.initiator;
+  else if (transfer.direction === 'received' && transfer.target === this.raiden.address)
+    role = ApiRole.target;
+  else if (transfer.direction === 'sent') role = ApiRole.mediator;
+  else return; // filter out duplicated 'received' side of mediated transfers
+  return {
+    channel_identifier: transfer.channelId.toString(),
+    initiator: transfer.initiator,
+    locked_amount: transfer.amount.toString(),
+    payment_identifier: transfer.paymentId.toString(),
+    role,
+    target: transfer.target,
+    token_address: transfer.token,
+    token_network_address: transfer.tokenNetwork,
+    transferred_amount: transfer.amount.sub(transfer.fee).toString(),
+  };
+}
 
-  router.get('/:tokenAddress', (_request: Request, response: Response) => {
-    response.status(404).send('Not implemented yet');
-  });
+async function getPendingTransfers(this: Cli, request: Request, response: Response) {
+  this.raiden.transfers$
+    .pipe(
+      filter((t) => !t.completed),
+      filter((t) => !request.params.tokenAddress || request.params.tokenAddress === t.token),
+      filter((t) => !request.params.partnerAddress || request.params.partnerAddress === t.partner),
```

Comment:
Nitpick: Why? Does this lead to ugly code formatting or what is the intention? Code gets incomparable often read then written. But I'm willed to accept it here... Just want to achieve some awareness for such tiny things. Event code navigation, highlighting of references, renaming etc are just bad with single character variable names.

Reply:
Fixed.